runtime-modules/working-group/src/tests/mod.rs

```diff
@@ -115,14 +115,17 @@ fn add_opening_fails_with_zero_reward() {
 fn add_opening_fails_with_insufficient_balance() {
     build_test_externalities().execute_with(|| {
         HireLeadFixture::default()
-            .with_initial_balance(<Test as Trait>::MinimumApplicationStake::get() + 1)
+            .with_initial_balance(<Test as Trait>::MinimumApplicationStake::get())
             .hire_lead();
 
-        let add_opening_fixture = AddOpeningFixture::default();
+        let add_opening_fixture = AddOpeningFixture::default().with_stake_policy(StakePolicy {
+            stake_amount: <Test as Trait>::MinimumApplicationStake::get() + 1,
+            leaving_unstaking_period: <Test as Trait>::MinUnstakingPeriodLimit::get() + 1,
+        });
 
-        add_opening_fixture.call_and_assert(Err(
-            Error::<Test, DefaultInstance>::InsufficientBalanceToCoverStake.into(),
-        ));
+        add_opening_fixture.call_and_assert(Err(DispatchError::Other(
+            "Not enough balance for a new stake.",
+        )));
```

Comment:
I see that the situation I described above is treated as expected test result here, but I think it shouldn't be for 2 reasons:
- `set_stake` is under the `MUTATION SAFE` mark, so we shouldn't expect it to ever throw
- `ensure_stake_for_opening_type` becomes useless at this point

Reply:
The `set_stake` method and changing the usable balance to free balance check are separate concerns. The previous `usable_balance` check was a bug. Even if the `set_stake` has issues we should fix them as well. But I don't see how it is the case. The `set_stake` method call is the first mutation method after the `mutation safe` mark on each occurrence and it returns the `DispatchResult` so it's potentially unsafe. But its first position (after the mark) grants us safety guarantees because if it fails the state remains unchanged. As to `ensure_stake_for_opening_type` it contains other checks in addition to balance verification but even if it would contain only the balance check for a stake - it's easier to reason about the `mutation safe` mark having all the checks set prior to mutation (even with embedded checks).